packages/ai-jsx/src/react/completion.tsx

```diff
@@ -100,5 +80,35 @@ export async function* UICompletion(
     return <Component>{children.map(toComponent)}</Component>;
   }
 
-  return <AI.React>{toComponent(JSON.parse(modelResult))}</AI.React>;
+  const Element: z.Schema = z.object({
+    name: z.string(),
+    children: z.array(z.union([z.string(), z.lazy(() => Element)])),
+  });
+
+  const modelRenderGenerator = render(
+    <JsonChatCompletion
+      schema={Element}
+      example={'<SomeComponent /> becomes: { "name": "SomeComponent", "children": [] }'}
```

Comment:
The `example` prop does seem a bit wishy-washy. Gotta think about a better name/framing for it.